src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,64 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+    private final int randomSite = random.nextInt(100) + 1;
+    private final int figureCount = random.nextInt(FIGURE_COUNT);
+    private final FigureList[] values = FigureList.values();
+    private final FigureList randomFigureType = values[figureCount];
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final String randomColor = colorSupplier.getRandomColor();
```

Comment:
There are many redundant fields
```suggestion
    public static final int FIGURE_COUNT = 5;
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```